### PR TITLE
218/jake/delegating validators part 2

### PR DESCRIPTION
**Pull request by jakegraves-figure**
Still a WIP with lots of things not hooked up yet. This branch is just getting large so I need to get it out of the way. If you have things you see that need fixing, let me know and I'll address them in future pull requests.

### CODE REVIEW

File: lib/screens/home/explorer/staking_delegation/staking_delegate.dart [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+  @override
+  void dispose() {
+    _textEditingController.removeListener(listen);
+    _textEditingController.dispose();
+
+    super.dispose();
+  }
+
+  void listen() {
```
Comment (royd) [inline lib/screens/home/explorer/staking_delegation/staking_delegate.dart:40]: `_onTextChanged` would be more descriptive

File: lib/screens/home/explorer/staking_delegation/staking_delegation_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  claimRewards,
+  undelegate,
+  redelegate,
+}
+
+extension SelectedModalTypeExtension on SelectedDelegationType {
+  String get dropDownTitle {
+    switch (this) {
+      case SelectedDelegationType.initial:
+        return "Back";
```
Comment (royd) [inline lib/screens/home/explorer/staking_delegation/staking_delegation_bloc.dart:149]: use strings file for these

File: lib/screens/home/explorer/staking_delegation/staking_redelegation_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import 'package:provenance_wallet/services/models/asset.dart';
+import 'package:provenance_wallet/services/models/delegation.dart';
+import 'package:provenance_wallet/services/models/detailed_validator.dart';
+import 'package:provenance_wallet/util/get.dart';
+import 'package:rxdart/rxdart.dart';
+
+// FIXME: Not used/registered yet.
+class StakingRedelegationBloc extends Disposable {
+  final BehaviorSubject<StakingDelegationDetails> _stakingDelegationDetails;
+  final _isLoading = BehaviorSubject.seeded(false);
```
Comment (royd) [inline lib/screens/home/explorer/staking_delegation/staking_redelegation_bloc.dart:17]: I would move these below the constructor for consistency

**Review by royd — APPROVED** [review]

File: lib/screens/home/explorer/staking_delegation/staking_delegation_bloc.dart (hunk 2 above)
Comment (jakegraves-figure) [reply lib/screens/home/explorer/staking_delegation/staking_delegation_bloc.dart:149]: Nice catch!

**Comment by jakegraves-figure** [comment]
@royd all the requested changes have been made in the next branch to be submitted for pull request.

**Pull request merged**